src/Libraries/Microsoft.Extensions.Http.Resilience/buildTransitive/Microsoft.Extensions.Http.Resilience.targets

```diff
@@ -0,0 +1,53 @@
+<Project>
+  <PropertyGroup>
+    <_GrpcNetClientFactory>Grpc.Net.ClientFactory</_GrpcNetClientFactory>
+    <_CompatibleGrpcNetClientFactoryVersion>2.64.0</_CompatibleGrpcNetClientFactoryVersion>
+    <_GrpcNetClientFactoryVersionIsIncorrect>Grpc.Net.ClientFactory 2.63.0 or earlier could cause issues when used together with Microsoft.Extensions.Http.Resilience. For more details, see https://learn.microsoft.com/en-us/dotnet/core/resilience/http-resilience#known-issues. Consider using Grpc.Net.ClientFactory $(_CompatibleGrpcNetClientFactoryVersion) or later. To suppress the error set SuppressCheckGrpcNetClientFactoryVersion=true.</_GrpcNetClientFactoryVersionIsIncorrect>
+  </PropertyGroup>
+
+  <!--
+    Check whether the project is referencing Grpc.Net.ClientFactory 2.64.0 or later.
+    If the vesion is earlier warn the user to update.
+    See https://learn.microsoft.com/en-us/dotnet/core/resilience/http-resilience#known-issues for more details.
+    -->
+  <Target Name="_CheckGrpcNetClientFactoryVersion"
+          BeforeTargets="ResolveReferences"
+          Condition=" '$(SuppressCheckGrpcNetClientFactoryVersion)' != 'true' ">
+    <ItemGroup>
+      <!-- Find the package in the .csproj file. -->
+      <_GrpcNetClientFactoryPackageReference Include="@(PackageReference)" Condition=" '%(PackageReference.Identity)' == '$(_GrpcNetClientFactory)' " />
+
+      <!-- Find the version of the package in the Central Package Source. The solution uses the Central Package Management. -->
+      <_GrpcNetClientFactoryPackageVersion Include="@(PackageVersion)" Condition=" '%(PackageVersion.Identity)' == '$(_GrpcNetClientFactory)' " />
+      
+      <!-- The package is added to the project as a transitive dependency. -->
+      <_GrpcNetClientFactoryTransitiveDependency Include="@(ReferencePath)" Condition=" '%(ReferencePath.NuGetPackageId)' == '$(_GrpcNetClientFactory)' " />
+    </ItemGroup>
+
+    <!-- The version of the package is included in the .csproj file. -->
+    <Error Condition=" @(_GrpcNetClientFactoryPackageReference->Count()) &gt; 0
```

Comment:
@RussKie @joperezr Are we good with notifying users about potential issues with Grpc.Net.ClientFactory as compilation errors? I think it should an error, since it is much easier to not notice a warning. What are thoughts on that? Thank you.

Reply:
We would typically avoid as much as possible introducing a build breaking change across minor versions unless it is absolutely required. I don't have a lot of background on the issue here, but from the error message it seems to be that while this could cause issues it will not always be the case, so just based on that I would suggest to consider using a Warning instead. A lot of people use Warn as errors anyway, so this would still be caught. I like the fact that this is suppressible. One suggestion would be to update the Warning text depending on whether the version is specified via PackageReference, PackageVersion, or ReferencePath, as that would help the user diagnose and understand how to update the version.

Reply:
I actually think it should be an "error" since using [..., 2.63] would result in a runtime error. We want actively stop the user at the compile-time.

Reply:
Would it always be a runtime error? From the above that didn't seem like the case.

Reply:
My interpretation is that there's a significant risk per the new readme. And we have two independent reports.
I suppose, we could make it a warning and monitor for any similar reports in the future - that could be a signal to up the severity.

Reply:
up to you and @iliar-turdushev as I trust you can assess the risk here, but in general as you know we really try to avoid introducing an intentional breaking change in a minor release

Reply:
Thank you for the feedback. Decided to go with Warning message, because the error occurs when a user enables the standard resilience or hedging handlers for a GrpcClient which is currently unsupported use-case. We think that users will not do that intentionally.